src/endpoint/s3/ops/s3_post_object_uploadId.js

```diff
@@ -24,27 +25,53 @@ async function post_object_uploadId(req, res) {
         throw new S3Error(S3Error.MalformedXML);
     }
 
-    const reply = await req.object_sdk.complete_object_upload({
-        obj_id: req.query.uploadId,
-        bucket: req.params.bucket,
-        key: req.params.key,
-        md_conditions: http_utils.get_md_conditions(req),
-        multiparts
-    });
+    // Write periodically to keep the connection alive
+    // TODO: Every complete above the S3_KEEP_ALIVE_WHITESPACE_INTERVAL
+    // will be responded in BODY, we cannot rollback from that for pre HEADER failures
+    // instread of having a fine tuned way start of sending body to make sure that we won't respond by BODY on HEADER failures
+    const interval_body_update = setInterval(() => {
+        if (res.headersSent) {
+            // Keep the connection alive with white spaces
+            res.write(' ');
+        } else {
+            // Mark begining of BODY response
+            // Assuming that write automatically assigns 200 OK status
+            res.write('<?xml version="1.0" encoding="UTF-8"?>');
+        }
+    }, config.S3_KEEP_ALIVE_WHITESPACE_INTERVAL);
 
-    s3_utils.set_encryption_response_headers(req, res, reply.encryption);
+    try {
+        const reply = await req.object_sdk.complete_object_upload({
+            obj_id: req.query.uploadId,
+            bucket: req.params.bucket,
+            key: req.params.key,
+            md_conditions: http_utils.get_md_conditions(req),
+            multiparts
+        });
 
-    if (reply.version_id && reply.version_id !== 'null') {
-        res.setHeader('x-amz-version-id', reply.version_id);
-    }
-    return {
-        CompleteMultipartUploadResult: {
-            Bucket: req.params.bucket,
-            Key: req.params.key,
-            ETag: `"${reply.etag}"`,
-            Location: req.originalUrl,
+        // TODO: Should be refactored when coding the complete solution
+        // We need to know all of these things prior to completion
+        // This also leaves a gap on long versioned/encrypted uploads (failure to assign headers after responding)
+        s3_utils.set_encryption_response_headers(req, res, reply.encryption);
+
+        if (reply.version_id && reply.version_id !== 'null') {
+            res.setHeader('x-amz-version-id', reply.version_id);
         }
-    };
+
+        return {
+            CompleteMultipartUploadResult: {
+                Bucket: req.params.bucket,
+                Key: req.params.key,
+                ETag: `"${reply.etag}"`,
+                Location: req.originalUrl,
+            }
+        };
+    } catch (error) {
+        if (res.headersSent) dbg.error('Error during long complete_object_upload', error);
+        throw error;
```

Comment:
I would move this logging to http_utils.send_reply

Reply:
Send_reply only works in the happy case when the function succeeds and doesn't throw.
I think that if we want this we can add later on inside the handle_error function in s3_rest, but I will leave it for now.